### PR TITLE
fix throwing

### PR DESCRIPTION
**Pull request by sethalves**
- when throwing something with hand-controllers, the release velocity should feel more natural.
- it's now less likely that, when throwing something, the thing will collide with your own avatar's collision capsule before it flies away.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 051616d7c3045d14132ae142453bf3066efc3ad5 available at: https://gist.github.com/96d0d16bada63224d3b12e873dbf87f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: eac144f35476762dfb6d99155320d22a00445b69 available at: https://gist.github.com/96d0d16bada63224d3b12e873dbf87f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-eac144f.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-eac144f.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7295e5c101fe1c0fe06b605f8087d7da53738059 available at: https://gist.github.com/96d0d16bada63224d3b12e873dbf87f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-7295e5c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-7295e5c.dmg

**Comment by sethalves** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7295e5c101fe1c0fe06b605f8087d7da53738059 available at: https://gist.github.com/96d0d16bada63224d3b12e873dbf87f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-7295e5c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-7295e5c.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c034879b740914a7496112a1094e9ce41edb28f9 available at: https://gist.github.com/96d0d16bada63224d3b12e873dbf87f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-c034879.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-c034879.dmg

File: interface/src/avatar/AvatarActionHold.cpp [hunk 1]
```diff
@@ -168,7 +168,7 @@ bool AvatarActionHold::getTarget(float deltaTimeStep, glm::quat& rotation, glm::
         position = palmPosition + rotation * _relativePosition;
 
         // update linearVelocity based on offset via _relativePosition;
-        linearVelocity = linearVelocity + glm::cross(angularVelocity, position - palmPosition);
+        // linearVelocity = linearVelocity + glm::cross(angularVelocity, position - palmPosition);
```
Comment (AndrewMeadows) [inline interface/src/avatar/AvatarActionHold.cpp]: No need to keep this if we're not using it anymore, so I guess we should purge both the commented lines.

File: interface/src/avatar/AvatarActionHold.cpp [hunk 2]
```diff
@@ -204,8 +204,17 @@ void AvatarActionHold::doKinematicUpdate(float deltaTimeStep) {
...
+            qDebug() << _measuredLinearVelocity.x << _measuredLinearVelocity.y << _measuredLinearVelocity.z
+                     << deltaTimeStep;
+        } else {
+            _measuredLinearVelocity = glm::vec3();
+        }
+
         if (_kinematicSetVelocity) {
-            rigidBody->setLinearVelocity(glmToBullet(_linearVelocityTarget));
+            // rigidBody->setLinearVelocity(glmToBullet(_linearVelocityTarget));
+            rigidBody->setLinearVelocity(glmToBullet(_measuredLinearVelocity));
```
Comment (AndrewMeadows) [inline interface/src/avatar/AvatarActionHold.cpp]: Similarly here: remove commented out cruft.

**Comment by sethalves** [comment]
Andrew -- this is all experiments so far.  I'm using a PR to get windows builds for the machine with the vive.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 53a366d4e882c8255aeabc45f8c327f3e9c5671d available at: https://gist.github.com/96d0d16bada63224d3b12e873dbf87f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-53a366d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-53a366d.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9c17890035422ecbc0bc0031ecc72f8d6e699c0f available at: https://gist.github.com/96d0d16bada63224d3b12e873dbf87f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-9c17890.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-9c17890.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b0f1492ac7b3844fd5440dc9542069bdfe687ea5 available at: https://gist.github.com/96d0d16bada63224d3b12e873dbf87f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-b0f1492.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-b0f1492.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1ef26d39eb43162f1fe5253841ab5ad56bf79a8c available at: https://gist.github.com/96d0d16bada63224d3b12e873dbf87f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-1ef26d3.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-1ef26d3.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0d0e0925e5b27f4944bc4cc074cf3e333da49386 available at: https://gist.github.com/96d0d16bada63224d3b12e873dbf87f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-0d0e092.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-0d0e092.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 84c5bef487fe6d16e38c3aa4e8ffd9b6f5802f8a available at: https://gist.github.com/96d0d16bada63224d3b12e873dbf87f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-84c5bef.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-84c5bef.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 62abe43996513271c860210b36006a93e93f4bc9 available at: https://gist.github.com/96d0d16bada63224d3b12e873dbf87f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-62abe43.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-62abe43.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6c14ef3392dbdbef2ffa577dabf567f22494d6c6 available at: https://gist.github.com/96d0d16bada63224d3b12e873dbf87f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-6c14ef3.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-6c14ef3.dmg

**Comment by sethalves** [comment]
to test:
- ready your hand controllers
- try throwing things
- try handing things to other people

**Comment by hifi-gustavo** [comment]
Build console output for commit: 19991939073353002b098df01a6d193e67890832 available at: https://gist.github.com/96d0d16bada63224d3b12e873dbf87f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-1999193.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-1999193.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0dd2171b759c1c135485501c32d1b80ada52354b available at: https://gist.github.com/96d0d16bada63224d3b12e873dbf87f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-0dd2171.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8217-0dd2171.dmg

**Comment by thoys** [comment]
## Testing report

- Had @kevinmthomasse with me as person that I hand the entities to.
- Tested both Hydras/Vive
- With the Hydra's it was a little more difficult to throw, but not impossible.
- With the Vive it was easy to throw things around
- Handing things to @kevinmthomasse went without problem. He even handed me back some objects.

QA Passed ( But the hydra's might need some more work in the future)

**Comment by sethalves** [comment]
Can you talk more about the trouble with the hydras?  I have hydras here and they seem okay, to me.

**Comment by thoys** [comment]
@sethalves , I was in the home, I tried to throw an apple, but it flung behind me several times, maybe I lack skills? :) (also the speed I put into it doesn't seem to translate while using the hydra's )

**Comment by thoys** [comment]
For reference, @sethalves came into my house to review my throwing bug, and after a few throws we decided that the Hydra's are not very reliable.

**Pull request merged**